app/models/book.rb

```diff
@@ -1,3 +1,60 @@
 class Book < ApplicationRecord
   has_and_belongs_to_many :tags
+  has_one :reservation, dependent: :destroy
+  belongs_to :user, foreign_key: "user_id"
+
+  validate :cannot_edit_reserved_or_booked
+  before_destroy :cannot_destroy_reserved_or_booked
+
+  STATUSES = [
+    AVAILABLE = 'available',
+    RESERVED = 'reserved',
+    BORROWED = 'borrowed'
+  ]
+
+  validates :status, :inclusion => { :in => STATUSES }, :allow_blank => false
+
+  def belongs_to?(current_user)
+  	user == current_user
```

Comment:
extra indentation here

Reply:
not sure why we'd call the parameter `current_user`, it's not really tied to the "current_user" and so I suggest making the name a bit more generic

Reply:
Ah, I was meaning to change that, this was just my temporary lack of imagination when trying to avoid writing `user == user`. I've updated this now to `self.user == user`.